featurebyte/models/request_input.py

```diff
@@ -148,20 +151,50 @@ async def materialize(
         """
         query_expr = self.get_query_expr(source_info=session.get_source_info())
 
+        # estimate the maximum POINT_IN_TIME value to filter the sample
+        column_names_and_dtypes = await self.get_column_names_and_dtypes(session=session)
+        available_columns = list(column_names_and_dtypes.keys())
+        if self.columns is None:
+            columns = available_columns
+        else:
+            columns = self.columns
+
+        if self.columns_rename_mapping is not None:
+            output_columns = [self.columns_rename_mapping.get(col, col) for col in columns]
+            column_names_and_dtypes = {
+                self.columns_rename_mapping.get(col, col): dtype
+                for col, dtype in column_names_and_dtypes.items()
+            }
+        else:
+            output_columns = columns
+
+        if SpecialColumnName.POINT_IN_TIME in output_columns and column_names_and_dtypes[
+            SpecialColumnName.POINT_IN_TIME
+        ] in {
+            DBVarType.TIMESTAMP,
+            DBVarType.TIMESTAMP_TZ,
+        }:
+            max_timestamp = datetime.utcnow() - timedelta(
+                hours=HISTORICAL_REQUESTS_POINT_IN_TIME_RECENCY_HOUR
+            )
```

Comment:
`RequestInput.materialize` is called not just when creating observation table, but also when creating batch request table and source static source table:

<img width="443" alt="image" src="https://github.com/user-attachments/assets/31275e84-9858-4c14-9d1f-4ebe5e0b0ee0">

The new logic here is only applicable when creating observation table.

Reply:
I think a cleaner approach would be to achieve this behaviour through parameters passed to `materialize` in observation table creation task:

* `sample_to_timestamp`: update this value to account for `HISTORICAL_REQUESTS_POINT_IN_TIME_RECENCY_HOUR` before calling the method
* `columns_exclude_missing_values`: List of columns that should not have missing values (we know what columns will be entity columns)